compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -224,11 +227,11 @@ FailureOr<ParameterSetting> ParameterSetting::create(
     uint32_t K0 = 1;
 
     // Instead of directly packing to (1, 1, M0, N0), the new strategy is making
-    // the pack size as (2, 2, M0/2, N0/2) to avoid the large allocation in L1.
+    // the pack size as (4, 4, M0/4, N0/4) to avoid the large allocation in L1.
     // Also we should make sure the first level inner pack size is divisible by
     // the second level of inner pack size (vector instruction size).
-    uint32_t m0Pack = (M0 / 2) % m1Pack == 0 ? (M0 / 2) : M0;
-    uint32_t n0Pack = (N0 / 2) % n1Pack == 0 ? (N0 / 2) : N0;
+    uint32_t m0Pack = (M0 / numCols) % m1Pack == 0 ? (M0 / numCols) : M0;
```

Comment:
`numCols == numRows` isn't a good assumption to make, especially as future architectures won't even have the same number of columns as rows.  At least call the variable `numRowsAndCols` to make it easier to unravel in the future. We should probably make the tiling here dependent on the number of rows and columns used (npu1_4col vs npu1_2col?)

Reply:
Okay, I'll change the variable name. I agree the tiling strategy needs to be changed based on the device info. This will be addressed as a follow up.